test/plugins/builds.test.js

```diff
@@ -1754,6 +1838,471 @@ describe('build plugin test', () => {
                     });
                 });
             });
+
+            describe('join new flow', () => {
```

Comment:
this is the tests for the new flow
current added tests are for internal join only
need to add more tests for external join workflow